cpp/cmake_modules/FindLz4.cmake

```diff
@@ -24,13 +24,14 @@ set(LZ4_STATIC_LIB_SUFFIX "${LZ4_MSVC_STATIC_LIB_SUFFIX}${CMAKE_STATIC_LIBRARY_S
 set(LZ4_LIB_NAMES lz4 liblz4)
 
 if(NOT ARROW_LZ4_USE_SHARED)
+  set(static_names_)
```

Comment:
This is not necessary.

Reply:
If someone defines `static_name_` before calling `FindLz4.cmake`, the `static_name_` value is used.
Is it wrong?